src/layout/umap.c

```diff
@@ -531,11 +532,8 @@ static igraph_error_t igraph_i_umap_compute_cross_entropy(const igraph_t *graph,
 
 
 /* clip forces to avoid too rapid shifts */
-static igraph_error_t igraph_i_umap_clip_force(igraph_real_t *force, igraph_real_t limit) {
-
+static void igraph_i_umap_clip_force(igraph_real_t *force, igraph_real_t limit) {
```

Comment:
Why not just return the result directly instead of through a pointer argument?

Reply:
```
static igraph_real_t igraph_i_umap_clip_force(igraph_real_t force, igraph_real_t limit) {
    if (force < -limit) return -limit;
    if (force > limit) return limit;
    return force;
}
```

This will also pass through any NaNs while being more efficient than the `fmin`/`fmax` version.

Reply:
Because there was no reason to change it. Now I did change it.